File: Vector.cpp
#include "Vector.h"
#include <cmath>

// Конструктор без параметрів
Vector::Vector()
{
    r = 0;
    angle = 0;
}

// Конструктор з параметрами
Vector::Vector(double r, double angle)
{
    this->r = r;
    this->angle = angle;
}

// Конструктор копіювання
Vector::Vector(const Vector& other)
{
    r = other.r;
    angle = other.angle;
}

// Деструктор
Vector::~Vector()
{
}

// Методи доступу
double Vector::getR() const
{
    return r;
}

double Vector::getAngle() const
{
    return angle;
}

// Обчислення X
double Vector::getX() const
{
    return r * cos(angle);
}

// Обчислення Y
double Vector::getY() const
{
    return r * sin(angle);
}

File: Vector.cs
using System;

namespace VectorLibrary
{
    public class Vector
    {
        // Закриті поля (інкапсуляція)
        private double r;      // довжина
        private double angle;  // кут у радіанах

        // Властивості тільки для читання
        public double R
        {
            get { return r; }
        }

        public double Angle
        {
            get { return angle; }
        }

        // 1. Конструктор без параметрів
        public Vector()
        {
            r = 0;
            angle = 0;
        }

        // 2. Конструктор з параметрами
        public Vector(double r, double angle)
        {
            this.r = r;
            this.angle = angle;
        }

        // 3. Конструктор копіювання
        public Vector(Vector other)
        {
            this.r = other.r;
            this.angle = other.angle;
        }

        // Метод обчислення X
        public double GetX()
        {
            return r * Math.Cos(angle);
        }

        // Метод обчислення Y
        public double GetY()
        {
            return r * Math.Sin(angle);
        }

        // Деструктор
        ~Vector()
        {
            // Виконується при знищенні об'єкта
        }
    }
}
